packages/avatar/src/vaadin-avatar-mixin.js

```diff
@@ -0,0 +1,240 @@
+         * The path to the image
+         */
+        img: {
+          type: String,
+          reflectToAttribute: true,
+          observer: '__imgChanged',
+        },
+
+        /**
+         * A shortened form of name that is displayed
+         * in the avatar when `img` is not provided.
+         */
+        abbr: {
+          type: String,
+          reflectToAttribute: true,
+        },
+
+        /**
+         * Full name of the user
+         * used for the tooltip of the avatar.
+         */
+        name: {
+          type: String,
+          reflectToAttribute: true,
+        },
+
+        /**
+         * Color index used for avatar background.
+         * @attr {number} color-index
+         */
+        colorIndex: {
+          type: Number,
+          observer: '__colorIndexChanged',
+        },
+
+        /**
+         * The object used to localize this component.
+         * To change the default localization, replace the entire
+         * _i18n_ object or just the property you want to modify.
+         *
+         * The object has the following JSON structure and default values:
+         *
+         * ```
+         * {
+         *   // Translation of the anonymous user avatar tooltip.
+         *   anonymous: 'anonymous'
+         * }
+         * ```
+         *
+         * @type {!AvatarI18n}
+         * @default {English/US}
+         */
+        i18n: {
+          type: Object,
+          value: () => {
+            return {
+              anonymous: 'anonymous',
+            };
+          },
+        },
+
+        /**
+         * When true, the avatar has tooltip shown on hover and focus.
+         * The tooltip text is based on the `name` and `abbr` properties.
+         * When neither is provided, `i18n.anonymous` is used instead.
+         * @attr {boolean} with-tooltip
+         */
+        withTooltip: {
+          type: Boolean,
+          value: false,
+          observer: '__withTooltipChanged',
+        },
+
+        /** @protected */
+        __imgVisible: Boolean,
+
+        /** @protected */
+        __iconVisible: Boolean,
+
+        /** @protected */
+        __abbrVisible: Boolean,
+
+        /** @private */
+        __tooltipNode: Object,
+      };
+    }
+
+    static get observers() {
+      return [
+        '__imgOrAbbrOrNameChanged(img, abbr, name)',
+        '__i18nChanged(i18n)',
+        '__tooltipChanged(__tooltipNode, name, abbr)',
+      ];
+    }
+
+    /** @protected */
+    ready() {
+      super.ready();
+
+      this.__updateVisibility();
+
+      // By default, if the user hasn't provided a custom role,
+      // the role attribute is set to "button".
+      if (!this.hasAttribute('role')) {
+        this.setAttribute('role', 'button');
+      }
+
+      if (!this.hasAttribute('tabindex')) {
+        this.setAttribute('tabindex', '0');
+      }
+
+      // Should set `anonymous` if name / abbr is not provided
+      if (!this.name && !this.abbr) {
+        this.__setTooltip();
```

Comment:
Not sure if this also qualifies for the inverse ready call-order in Lit components we discussed today? Then the tooltip might not be initialized here yet. Tests didn't fail in my prototype though, assuming it is covered.

Reply:
Actually the tooltip is not rendered via Lit, but manually created in an observer and then also updated, so shouldn't be a problem.